pkg/localkube/localkube.go

```diff
@@ -17,61 +17,90 @@ limitations under the License.
 package localkube
 
 import (
-	"errors"
 	"fmt"
-	"io"
+	"net"
+	"path"
+
+	utilcrypto "k8s.io/kubernetes/pkg/util"
+	utilnet "k8s.io/kubernetes/pkg/util/net"
+
+	"k8s.io/minikube/pkg/util"
 )
 
-// LocalKube provides a fully functional Kubernetes cluster running entirely through goroutines
-type LocalKube struct {
+const serverInterval = 200
+
+// LocalkubeServer provides a fully functional Kubernetes cluster running entirely through goroutines
+type LocalkubeServer struct {
+	// Inherits Servers
 	Servers
+
+	// Options
+	Containerized            bool
+	EnableDNS                bool
+	DNSDomain                string
+	DNSIP                    net.IP
+	LocalkubeDirectory       string
+	ServiceClusterIPRange    net.IPNet
+	APIServerAddress         net.IP
+	APIServerPort            int
+	APIServerInsecureAddress net.IP
+	APIServerInsecurePort    int
 }
 
-func (lk *LocalKube) Add(server Server) {
+func (lk *LocalkubeServer) AddServer(server Server) {
 	lk.Servers = append(lk.Servers, server)
 }
 
-func (lk *LocalKube) Run(args []string, out io.Writer) error {
-	if len(args) < 1 {
-		return errors.New("you must choose start <name>, stop <name>, or status")
+func (lk LocalkubeServer) GetEtcdDataDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "etcd")
+}
+
+func (lk LocalkubeServer) GetDNSDataDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "dns")
+}
+
+func (lk LocalkubeServer) GetCertificateDirectory() string {
+	return path.Join(lk.LocalkubeDirectory, "certs")
+}
+func (lk LocalkubeServer) GetPrivateKeyCertPath() string {
+	return path.Join(lk.GetCertificateDirectory(), "apiserver.key")
+}
+func (lk LocalkubeServer) GetPublicKeyCertPath() string {
+	return path.Join(lk.GetCertificateDirectory(), "apiserver.crt")
+}
+
+func (lk LocalkubeServer) GetAPIServerSecureURL() string {
+	return fmt.Sprintf("https://%s:%d", lk.APIServerAddress.String(), lk.APIServerPort)
+}
+
+func (lk LocalkubeServer) GetAPIServerInsecureURL() string {
+	return fmt.Sprintf("http://%s:%d", lk.APIServerInsecureAddress.String(), lk.APIServerInsecurePort)
+}
+
+// Get the host's public IP address
+func (lk LocalkubeServer) GetHostIP() (net.IP, error) {
+	return utilnet.ChooseBindAddress(net.ParseIP("0.0.0.0"))
```

Comment:
Two files are created, apiserver.crt, which functions both as public key as well as ca. This cert should contain all ips as you suggested. The apiserver.key file serves as the private key. Not sure what you meant with the last comment...